lib/posix/pthread.c

```diff
@@ -459,6 +459,18 @@ int pthread_create(pthread_t *th, const pthread_attr_t *_attr, void *(*threadrou
 	return 0;
 }
 
+int pthread_getconcurrency(void)
+{
+	return 0;
```

Comment:
how is this an implementation? Doesn't it make more sense to leave this as Not supported and not implement it this way? If I have the API available and I can call it, the expectation is that it will work as specified...

Reply:
@nashif From https://pubs.opengroup.org/onlinepubs/9699919799/functions/pthread_getconcurrency.html
> APPLICATION USAGE
> Application developers should note that an implementation can always ignore any calls to pthread_setconcurrency() and return a constant for pthread_getconcurrency(). For this reason, it is not recommended that portable applications use this function.

So, it technically does work as specified.

I'm not 100% sure if the alternative is possible today in Zephyr (i.e. adjusting at runtime the number of active threads to be less than the number of cores / harts). So it would be a long rabbit hole to go down.

Reply:
Also, POSIX requires that all components of a particular option group are supported if the implementation is to claim that the option group is supported. Alternatively, if they are to be unimplemented, the symbols still need to be present, but docs must state that they fall into the category of undefined behaviour.

These were the only two missing functions from `XSI_THREADS_EXT`. The trivial implementation conforms to the spec as well, so there is no need to document any kind of undefined behaviour.

<img width="612" alt="Screenshot 2023-11-23 at 10 20 39 PM" src="https://github.com/zephyrproject-rtos/zephyr/assets/983494/da44efda-c118-417d-a64a-af0456f5accb">

If Zephyr is to one day support the functionality for adjusting concurrency dynamically, then we can revisit, but that will not be anytime soon.

Reply:
If you are going to add this implementation that does not have any effect you should not document this as supported. You need to say that this has no effect and implemented for completeness and source code compatbility. This is not the case now. See an example of how this is done elsewhere:

https://www.qnx.com/developers/docs/6.6.0_anm11_wf10/#com.qnx.doc.neutrino.lib_ref/topic/p/pthread_getconcurrency.html

> The .... RTOS doesn't support the multiplexing of user threads on top of several kernel scheduled entities. As such, the [pthread_setconcurrency()](https://www.qnx.com/developers/docs/6.6.0_anm11_wf10/com.qnx.doc.neutrino.lib_ref/topic/p/pthread_setconcurrency.html) and pthread_getconcurrency() functions are provided for source code compatibility but they have no effect when called. To maintain the function semantics, the new_level parameter is saved when pthread_setconcurrency() is called so that a subsequent call to pthread_getconcurrency() returns the same value.

Reply:
Again - reworking posix docs are out of the scope of this PR.

Reply:
I am not asking you to rework the documentation here. 

The change to the documentation related to those calls however is wrong. This is not supported and misleading to say this function is supported when you are just adding it for source code compatibility. So please remove the 'supported' and come up with a new way to document such functions without reworking the docs or add a footnote a section somewhere indicating this and similar calls hav no effect etc and implemented for source code compatibility only.